subiquity/controllers/installprogress.py

```diff
@@ -307,76 +452,65 @@ def _bg_cleanup_apt(self):
-            def __init__(self, controller):
-                self.controller = controller
-
-            def task_complete(self, stage):
-                pass
-
-            def task_error(self, stage, info):
-                if isinstance(info, tuple):
-                    tb = traceback.format_exception(*info)
-                    self.controller.curtin_error("".join(tb))
-                else:
-                    self.controller.curtin_error()
-
-            def tasks_finished(self):
-                self.controller.loop.set_alarm_in(
-                    0.0,
-                    lambda loop, ud: self.controller.postinstall_complete())
-        tasks = [
-            ('drain', WaitForCurtinEventsTask(self)),
-            ('cloud-init', InstallTask(
-                self, "configuring cloud-init",
-                self.base_model.configure_cloud_init)),
-        ]
-        if self.base_model.ssh.install_server:
-            tasks.extend([
-                ('install-ssh', InstallTask(
-                    self, "installing OpenSSH server",
-                    self._bg_install_openssh_server)),
-                ])
-        tasks.extend([
-            ('cleanup', InstallTask(
-                self, "restoring apt configuration",
-                self._bg_cleanup_apt)),
-            ])
-        ts = TaskSequence(self.run_in_bg, tasks, w(self))
-        ts.run()
+            self._bg_run_command_logged(cmd, check=True)
 
+    @task
     def postinstall_complete(self):
         self._install_event_finish()
         self.ui.set_header(_("Installation complete!"))
         self.progress_view.set_status(_("Finished install!"))
         self.progress_view.show_complete()
 
-        if self.answers['reboot']:
-            self.reboot()
-
-    def copy_logs_to_target(self):
+    @task(label="downloading and installing security updates",
+          transitions={'success': 'wait_for_click', 'reboot': 'abort_uu'},
+          net_only=True)
+    def _bg_run_uu(self):
         if self.opts.dry_run:
-            return
-        target_logs = self.tpath('var/log/installer')
-        utils.run_command(['cp', '-aT', '/var/log/installer', target_logs])
-        try:
-            with open(os.path.join(target_logs,
-                                   'installer-journal.txt'), 'w') as output:
-                utils.run_command(
-                    ['journalctl'],
-                    stdout=output, stderr=subprocess.STDOUT)
-        except Exception:
-            log.exception("saving journal failed")
+            self.uu = utils.start_command(["sleep", str(10/self.scale_factor)])
+            self.uu.wait()
+        else:
+            self._bg_run_command_logged([
+                sys.executable, "-m", "curtin", "in-target", "-t", "/target",
+                "--", "unattended-upgrades", "-v",
+            ], check=True)
+
+    @task(net_only=True)
+    def abort_uu(self):
+        self._install_event_finish()
 
+    @task(label="deferring update", net_only=True)
```

Comment:
I think "deferring" is unclear.  I suggest "cancelling installation of security updates".